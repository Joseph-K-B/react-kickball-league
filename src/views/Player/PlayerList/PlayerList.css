.listTitle {
  font-size: 2.25rem;
  text-align: center;
}

.playerList {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    width: 90%;
    text-align: center;
    align-items: center;
  }

  .playerListItem {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr 1fr;
    justify-content: center;
    margin: 2.5%;
    text-align: center;
    border: #F0EBD8 solid 0.15rem;
    border-radius: 7%;
  }