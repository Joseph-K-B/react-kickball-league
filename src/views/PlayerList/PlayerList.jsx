import { useEffect, useState } from "react"
import { Link } from "react-router-dom"
import { getPlayers, deletePlayerById } from "../../services/players"

function PlayerList() {
    const[loading, setLoading] = useState(true)
    const [players, setPlayers] = useState([])

    const reloadPlayers = async () => {
        setLoading(true);
        const res = await getPlayers();
        setPlayers(res);
        setLoading(false);
    }

    useEffect(() => {
        getPlayers()
        .then((res) => setPlayers(res))
        .finally(() => setLoading(false))
    }, []);

    const handlePlayerDelete = async ({id, name}) => {
        // eslint-disable-next-line no-restricted-globals
        const confirmPlayerDelete = confirm(`Would you like to remove ${name} from this seasons roster?`)

        if(confirmPlayerDelete) {
            await deletePlayerById(id);
            await reloadPlayers();
        }
    };

    if(loading) return <h1>Loading players...</h1>
    console.log(players)

    return(
        <>
        <h1>Player List</h1>
        <ul>
            {players.map((player) => (
                <li key={player.id}>
                    <Link className='link' to={`/players/${player.id}`}>
                        <p>{player.name}</p>
                    </Link>
                        <button
                            type="button"
                            aria-label={`Delete ${player.name}`}
                            onClick={() => handlePlayerDelete({ id: player.id, name: player.name})}
                        >
                        Delete
                    </button>
                    <Link to={`/players/${player.id}/update`}>
                        <button
                            type="button"
                            aria-label={`Update ${player.name}`}
                        >
                            Update
                        </button>
                    </Link>
                </li>
            ))}
        </ul>
        <Link className='link' to='/players/new'><p>Add Player</p></Link>
        </>
    )
}

export default PlayerList