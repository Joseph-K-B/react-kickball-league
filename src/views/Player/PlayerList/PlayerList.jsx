import { useEffect, useState } from "react"
import { Link } from "react-router-dom"
import { deletePlayerById, getPlayers } from "../../../services/players"
import "./PlayerList.css";

function PlayerList() {
    const[loading, setLoading] = useState(true)
    const [players, setPlayers] = useState([])

    const reloadPlayers = async () => {
        setLoading(true);
        const res = await getPlayers();
        setPlayers(res);
        setLoading(false);
    }

    useEffect(() => {
        getPlayers()
        .then((res) => setPlayers(res))
        .finally(() => setLoading(false))
    }, []);

    const handlePlayerDelete = async ({id, name}) => {
        // eslint-disable-next-line no-restricted-globals
        const confirmPlayerDelete = confirm(`Would you like to remove ${name} from this seasons roster?`)

        if(confirmPlayerDelete) {
            await deletePlayerById(id);
            await reloadPlayers();
        }
    };

    if(loading) return <h1>Loading players...</h1>
    
    return(
        <>
        <h1 className='listTitle'>Players</h1>
        <ul className='playerList'>
            {players.map((player) => (
                <li className='playerListItem' key={player.id}>
                    <Link className='link' to={`/players/${player.id}`}>
                        <p>{player.name}</p>
                    </Link>
                        <div>
                        <button
                            type="button"
                            aria-label={`Delete ${player.name}`}
                            onClick={() => handlePlayerDelete({ id: player.id, name: player.name})}
                            >
                            Delete
                        </button>
                        <Link to={`/players/${player.id}/update`}>
                            <button
                                type="button"
                                aria-label={`Update ${player.name}`}
                                >
                                Update
                            </button>
                        </Link>
                    </div>
                </li>
            ))}
        </ul>
        <Link className='link' to='/players/new'><p>Add Player</p></Link>
        </>
    )
}

export default PlayerList