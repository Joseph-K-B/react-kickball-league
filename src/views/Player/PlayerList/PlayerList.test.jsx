import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import PlayerList from './PlayerList'


it('renders team details according to id param', async () => {
    const {container} = render(
        <MemoryRouter initialEntries={['/players']}>
            <Route path="/players">
                <PlayerList />
            </Route>
        </MemoryRouter>
    );

    screen.getByText('Loading players...')

    const player = await screen.findByText(/Bill Waits/i);
    const playerTwo = await screen.findByText(/Izzie the Dog/i);
    const playerThree = await screen.findByText(/Kurt Coplain/i);
    expect(player).toBeInTheDocument();
    expect(playerTwo).toBeInTheDocument();
    expect(playerThree).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})