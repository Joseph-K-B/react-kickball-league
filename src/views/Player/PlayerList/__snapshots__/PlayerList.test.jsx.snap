// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team details according to id param 1`] = `
<div>
  <h1
    class="listTitle"
  >
    Players
  </h1>
  <ul
    class="playerList"
  >
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/8"
      >
        <p>
          Bill Waits
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Bill Waits"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/8/update"
        >
          <button
            aria-label="Update Bill Waits"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/4"
      >
        <p>
          Dangalf the Wise
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Dangalf the Wise"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/4/update"
        >
          <button
            aria-label="Update Dangalf the Wise"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/18"
      >
        <p>
          Donald Bump
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Donald Bump"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/18/update"
        >
          <button
            aria-label="Update Donald Bump"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/19"
      >
        <p>
          Emily Bunt
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Emily Bunt"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/19/update"
        >
          <button
            aria-label="Update Emily Bunt"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/10"
      >
        <p>
          Homerun Simpson
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Homerun Simpson"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/10/update"
        >
          <button
            aria-label="Update Homerun Simpson"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/11"
      >
        <p>
          Izzie the Dog
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Izzie the Dog"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/11/update"
        >
          <button
            aria-label="Update Izzie the Dog"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/3"
      >
        <p>
          JK Bowling
        </p>
      </a>
      <div>
        <button
          aria-label="Delete JK Bowling"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/3/update"
        >
          <button
            aria-label="Update JK Bowling"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/5"
      >
        <p>
          Jaba the Hook
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Jaba the Hook"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/5/update"
        >
          <button
            aria-label="Update Jaba the Hook"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/9"
      >
        <p>
          Jimi Spindrix
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Jimi Spindrix"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/9/update"
        >
          <button
            aria-label="Update Jimi Spindrix"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/7"
      >
        <p>
          Kurt Coplain
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Kurt Coplain"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/7/update"
        >
          <button
            aria-label="Update Kurt Coplain"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/1"
      >
        <p>
          Margaret Catcher
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Margaret Catcher"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/1/update"
        >
          <button
            aria-label="Update Margaret Catcher"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/6"
      >
        <p>
          Marty McWhy
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Marty McWhy"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/6/update"
        >
          <button
            aria-label="Update Marty McWhy"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/15"
      >
        <p>
          Nikki Kicks
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Nikki Kicks"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/15/update"
        >
          <button
            aria-label="Update Nikki Kicks"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/21"
      >
        <p>
          Ruby the Dog
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Ruby the Dog"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/21/update"
        >
          <button
            aria-label="Update Ruby the Dog"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/16"
      >
        <p>
          Ruth Batter Winsburg
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Ruth Batter Winsburg"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/16/update"
        >
          <button
            aria-label="Update Ruth Batter Winsburg"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/13"
      >
        <p>
          Water Guy
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Water Guy"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/13/update"
        >
          <button
            aria-label="Update Water Guy"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/2"
      >
        <p>
          Will I Aint
        </p>
      </a>
      <div>
        <button
          aria-label="Delete Will I Aint"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/2/update"
        >
          <button
            aria-label="Update Will I Aint"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
    <li
      class="playerListItem"
    >
      <a
        class="link"
        href="/players/22"
      >
        <p>
          random
        </p>
      </a>
      <div>
        <button
          aria-label="Delete random"
          type="button"
        >
          Delete
        </button>
        <a
          href="/players/22/update"
        >
          <button
            aria-label="Update random"
            type="button"
          >
            Update
          </button>
        </a>
      </div>
    </li>
  </ul>
  <a
    class="link"
    href="/players/new"
  >
    <p>
      Add Player
    </p>
  </a>
</div>
`;
